hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -443,23 +444,22 @@ public void onInvalidate(Invalidation invalidation) {
         }
 
         @Override
-        public void handle(BatchNearCacheInvalidation batchNearCacheInvalidation) {
-            List<SingleNearCacheInvalidation> invalidations = batchNearCacheInvalidation.getInvalidations();
+        public void handle(BatchNearCacheInvalidation batch) {
+            List<Invalidation> invalidations = batch.getInvalidations();
 
-            for (SingleNearCacheInvalidation invalidation : invalidations) {
-                handle(invalidation);
+            for (Invalidation invalidation : invalidations) {
+                invalidation.consumedBy(this);
```

Comment:
The main reason for using `invalidationHandler` is to avoid `instanceOf` checks since it is fragile here. By doing it this way, `invalidationHandler` won't be compiled when there is a new `invalidation` type introduced and it will force us to do required changes on relevant code parts. And yes that can cost indirections.